<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <!-- Meta tags for social media banners, these should be filled in appropriatly as they are your "business card" -->
    <!-- Replace the content tag with appropriate information -->
    <meta name="description" content="DeepChopper: A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing">
    <meta property="og:title" content="DeepChopper: A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing" />
    <meta property="og:description" content="DeepChopper: A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing" />
    <meta property="og:url" content="URL OF THE WEBSITE" />
    <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X630-->
    <meta property="og:image" content="static/image/your_banner_image.png" />
    <meta property="og:image:width" content="1200" />
    <meta property="og:image:height" content="630" />


    <meta name="twitter:title" content="DeepChopper: A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing">
    <meta name="twitter:description" content="DeepChopper: A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing">
    <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X600-->
    <meta name="twitter:image" content="static/images/your_twitter_banner_image.png">
    <meta name="twitter:card" content="summary_large_image">
    <!-- Keywords for your paper to be indexed by-->
    <meta name="keywords" content="Genomic Language Model, Nanopore Direct RNA, Chimera Artifact">
    <meta name="viewport" content="width=device-width, initial-scale=1">


    <title>A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing</title>
    <link rel="icon" type="image/x-icon" href="static/images/favicon.ico">
    <link href="https://fonts.googleapis.com/css?family=Google+Sans|Noto+Sans|Castoro" rel="stylesheet">

    <link rel="stylesheet" href="static/css/bulma.min.css">
    <link rel="stylesheet" href="static/css/bulma-carousel.min.css">
    <link rel="stylesheet" href="static/css/bulma-slider.min.css">
    <link rel="stylesheet" href="static/css/fontawesome.all.min.css">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons@1/css/academicons.min.css">
    <link rel="stylesheet" href="static/css/index.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="https://documentcloud.adobe.com/view-sdk/main.js"></script>
    <script defer src="static/js/fontawesome.all.min.js"></script>
    <script src="static/js/bulma-carousel.min.js"></script>
    <script src="static/js/bulma-slider.min.js"></script>
    <script src="static/js/index.js"></script>
</head>

<body>

    <section class="hero">
        <div class="hero-body">
            <div class="container is-max-desktop">
                <div class="columns is-centered">
                    <div class="column has-text-centered">
                        <h1 class="title is-1 publication-title">A Genomic Language Model for Chimera Artifact Detection in Nanopore Direct RNA Sequencing</h1>
                        <div class="is-size-5 publication-authors">
                            <!-- Paper authors -->
                            <span class="author-block">
                                <a href="https://yangyangli.top/about/" target="_blank">Yangyang Li</a><sup>*</sup>,</span>
                            <span class="author-block">
                                <a href="https://labs.feinberg.northwestern.edu/yang/members/index.html" target="_blank">Ting-You
                                    Wang</a><sup>*</sup>,</span>
                            <span class="author-block"></span>
                            <a href="https://labs.feinberg.northwestern.edu/yang/members/index.html" target="_blank">Qingxiang Guo</a>,
                            <a href="https://labs.feinberg.northwestern.edu/yang/members/index.html" target="_blank">Yanan Ren</a>,
                            <a href="https://labs.feinberg.northwestern.edu/yang/members/index.html" target="_blank">Xiaotong Lu</a>,
                            <a href="AUTHOR PERSONAL LINK" target="_blank">Qi Cao</a>,
                            <a href="https://labs.feinberg.northwestern.edu/yang/index.html" target="_blank">Rendong Yang</a>
                            </span>
                        </div>

                        <div class="is-size-5 publication-authors">
                            <span class="author-block">Northwestern University<br></span>
                            <span class="eql-cntrb"><small><br><sup>*</sup>Indicates Equal Contribution</small></span>
                        </div>

                        <div class="column has-text-centered">
                            <div class="publication-links">
                                <!-- Arxiv PDF link -->
                                <span class="link-block">
                                    <a href="https://www.biorxiv.org/content/10.1101/2024.10.23.619929v2.full.pdf"
                                        target="_blank" class="external-link button is-normal is-rounded is-dark">

                                        <span class="icon">
                                            <i class="fas fa-file-pdf"></i>
                                        </span>
                                <span>Paper</span>
                                </a>
                                </span>

                                <!-- Supplementary PDF link -->
                                <span class="link-block">
                                    <!-- <a href="static/pdfs/supplementary_material.pdf" target="_blank" -->
                                    <!-- class="external-link button is-normal is-rounded is-dark"> -->
                                    <!-- <span class="icon"> -->
                                    <!-- <i class="fas fa-file-pdf"></i> -->
                                    <!-- </span> -->
                                <!-- <span>Supplementary</span> -->
                                <!-- </a> -->
                                <!-- </span> -->

                                <!-- Github link -->
                                <span class="link-block">
                                        <a href="https://github.com/ylab-hi/DeepChopper" target="_blank"
                                            class="external-link button is-normal is-rounded is-dark">
                                            <span class="icon">
                                                <i class="fab fa-github"></i>
                                            </span>
                                <span>Code</span>
                                </a>
                                </span>

                                <!-- ArXiv abstract Link -->
                                <span class="link-block">

                                        <a href="https://www.biorxiv.org/content/10.1101/2024.10.23.619929v2"
                                            target="_blank" class="external-link button is-normal is-rounded is-dark">
                                            <span class="icon">
                                                <i class="ai ai-biorxiv"></i>
                                            </span>
                                <span>bioRxiv</span>
                                </a>

                                <!-- Hugging Face Space Link -->
                                <span class="link-block">
                                            <a href="https://huggingface.co/spaces/yangliz5/deepchopper" target="_blank"
                                                class="external-link button is-normal is-rounded is-dark">
                                                <span class="icon">
                                                    <i class="fab fa-hubspot"></i>
                                                </span>
                                <span>Demo</span>
                                </a>
                                </span>

                                </span>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </section>


    <!-- Teaser video-->
    <section class="hero teaser">
        <div class="container is-max-desktop">
            <div class="hero-body">
                <img src="static/images/image1.png" alt="MY ALT TEXT" style="transform: scale(1.3); transform-origin: center;" />
                <!-- <h2 class="subtitle has-text-centered"> -->
                <!-- Overview of the architecture of the DeepChopper model. -->
                <!-- </h2> -->
            </div>
        </div>
    </section>
    <!-- End teaser video -->

    <!-- Paper abstract -->
    <section class="section hero is-light">
        <div class="container is-max-desktop">
            <div class="columns is-centered has-text-centered">
                <div class="column is-four-fifths">
                    <h2 class="title is-3">Abstract</h2>
                    <div class="content has-text-justified">
                        <p>
                            Chimera artifacts in nanopore direct RNA sequencing (dRNA-seq) can significantly distort transcriptome analyses, yet their detection and removal remain challenging due to limitations in existing basecalling models. We present DeepChopper, a genomic language
                            model that precisely identifies and removes adapter sequences from base-called dRNA-seq long reads at single-base resolution, operating independently of raw signal or alignment information to effectively eliminate chimeric
                            read artifacts. By removing these artifacts, DeepChopper substantially improves the accuracy of critical downstream analyses, such as transcript annotation and gene fusion detection, thereby enhancing the reliability and utility
                            of nanopore dRNA-seq for transcriptomics research.
                        </p>
                    </div>
                </div>
            </div>
        </div>
    </section>
    <!-- End paper abstract -->


    <!-- Image carousel -->
    <section class="hero is-small">
        <div class="hero-body">
            <div class="container">
                <div id="results-carousel" class="carousel results-carousel">
                    <div class="item">
                        <!-- Your image here -->
                        <img src="static/images/image2.png" alt="MY ALT TEXT" />
                        <h2 class="subtitle has-text-centered">
                            <div class="content" style="text-align: justify;">
                                <strong>Detection of chimeric read artifacts in dRNA-seq data using DeepChopper and
                                    validation with orthogonal sequencing platforms</strong>
                                <p><strong>(b)</strong> Length distribution of predicted adapters by DeepChopper in VCaP dRNA-seq data.</p>
                                <p><strong>(c)</strong> Distribution of relative adapter position along read length in VCaP dRNA-seq data. Grey rectangle represents a long read from 5′ to 3′. Relative position is calculated as the ratio of the length before
                                    DeepChopper predicted adapter start position to the total read length.</p>
                                <p><strong>(d)</strong> Distribution of segments per read after trimming: 1 segment indicates 3′ end adapter trimmed, while 2 or more indicate internal adapters trimmed.
                                </p>
                                <p><strong>(e)</strong> Chimeric alignments (in thousands) for VCaP dRNA-seq reads processed by Dorado with and without adapter trimming, and DeepChopper. DeepChopper greatly reduces chimeric alignments not supported by direct
                                    cDNA sequencing.</p>
                                <p><strong>(f)</strong> Distribution of base qualities from identified internal adapters by DeepChopper. Background colors indicate quality levels: green (high), yellow (medium), and red (low).</p>
                                <p><strong>(g)</strong> BLAST-like alignment tool (BLAT) identity distribution of the internal adapter sequences mapping against human reference genome.</p>
                                <p><strong>(h)</strong> The number of chimeric alignments (in thousands) for A549, HCT116, HepG2, K562, and MCF7 cell lines processed by Dorado with adapter trimming and DeepChopper. DeepChopper consistently reduces chimeric
                                    alignments not supported by cDNA sequencing across all cell lines.</p>
                                <p><strong>(i)</strong> Chimeric alignments from dRNA-seq of the WTC11 cell line, evaluated for support using additional ONT and Pacific Biosciences (PacBio) sequencing data with different protocols. DeepChopper reduces unsupported
                                    chimeric alignments across all methods compared to Dorado with adapter trimming.</p>
                            </div>
                        </h2>
                    </div>
                    <div class="item">
                        <!-- Your image here -->
                        <img src="static/images/image3.png" alt="MY ALT TEXT" />
                        <h2 class="subtitle has-text-centered">
                            <div class="content" style="text-align: justify;">
                                <strong>Characterization of dRNA-seq chimera artifacts and their impact on downstream
                                    analysis in VCaP cells</strong>
                                <p><strong>(a)</strong> The upper bar plot shows the number of chimeric connections per kilobase across chromosomes, highlighting higher chimeric activity in Chr 19 and Chr M. The lower heatmap visualizes interchromosomal
                                    connections, with intensity indicating the count of connections between different chromosomes.
                                </p>
                                <p><strong>(b)</strong> The bar plot shows the number of transcripts (in thousands) across different isoform classification categories. DeepChopper-processed reads result in a higher number of transcripts compared to Dorado-trimmed
                                    reads. The inset details the isoform classification scheme.</p>
                                <p><strong>(c)</strong> Detected gene fusions from Dorado adapter-trimmed reads and DeepChopper-processed reads. Gene fusions identified from short-read RNA-seq were used to validate fusion events detected from dRNA-seq.</p>
                                <p><strong>(d)</strong> Gene Ontology (GO) enrichment analysis of chimera artifact-affected genes, with color indicating gene count per term.</p>
                                <p><strong>(e)</strong> Analysis of a chimeric read artifact detected as an RPS29-COX8A fusion. The schematic shows the fusion between RPS29 (Chr 14) and COX8A (Chr 11). The green plot indicates quality scores along the read,
                                    and the blue plot shows raw signal intensity (in pA). The chopped region identified by DeepChopper corresponds to a low-quality segment with low current intensity, polyA, and short open pore signals, suggesting the
                                    presence of an ONT adapter.</p>
                            </div>
                        </h2>
                    </div>

                    <div class="item">
                        <img src="static/images/sf1.png" alt="MY ALT TEXT" />
                        <h2 class="subtitle has-text-centered">
                            <div class="content" style="text-align: justify;">
                                <strong>Benchmarking of DeepChopper predictions on chimeric read artifacts using
                                    synthetic data and dRNA-seq data generated with the SQK-RNA004 kit from the VCaP
                                    cell line</strong>
                                <p><strong>(a)</strong> Performance evaluation in a held-out test dataset showing Recall, Precision, and F1 values for DeepChopper.</p>
                                <p><strong>(b)</strong> Number of chimeric alignments (in thousands) identified in VCaP RNA004 dRNA-seq reads processed by Dorado with and without adapter trimming, and by DeepChopper. The bar colors indicate chimeric reads
                                    supported by cDNA sequencing (red) and those lacking support (grey).</p>
                                <p><strong>(c)</strong> Base quality scores (left) and BLAT alignment identity scores (right) for internal adapter sequences identified by DeepChopper in RNA004 dRNA-seq reads.
                                </p>
                            </div>
                        </h2>
                    </div>

                    <div class="item">
                        <img src="static/images/sf2.png" alt="MY ALT TEXT" />
                        <h2 class="subtitle has-text-centered">
                            <div class="content" style="text-align: justify;">
                                <strong>Analysis of dRNA-seq chimera artifacts and their genomic and transcriptomic
                                    characteristics in VCaP cells</strong>
                                <p><strong>(a)</strong> Box plot comparing gene expression levels between all expressed genes (N=19,156) and genes affected by chimera artifacts (N=7,186) in the VCaP dRNA-seq dataset. Chimera artifacts-affected genes exhibit
                                    significantly higher expression levels (p-value
                                    < 2.2 × 10<sup>-16</sup>).
                                </p>
                                <p><strong>(b)</strong> Distribution of gene effective sizes for all expressed genes and genes affected by chimera artifacts, indicating that the size distributions of genes impacted by chimera artifacts are comparable to
                                    those of all expressed genes.</p>
                                <p><strong>(c)</strong> Chromosomal distribution and interchromosomal connections from chimeric read artifacts arising from VCaP RNA004 dRNA-seq. The top bar plot shows the number of connections per kilobase for each chromosome,
                                    with higher bars indicating more frequent connections. The bottom heatmap visualizes the number of chimeric connections between chromosome pairs, with color intensity representing the connection frequency.</p>
                                <p><strong>(d)</strong> Number of detected transcripts across different isoform categories (FSM, ISM, NIC, NNC, and Other) from DeepChopper-identified chimeric read artifacts in VCaP RNA004 dRNA-seq data. DeepChopper-corrected
                                    reads resulted in a greater number of transcripts compared to adapter-trimmed reads by Dorado across all categories.</p>
                            </div>
                        </h2>
                    </div>


                    <div class="item">
                        <img src="static/images/sf31.png" alt="MY ALT TEXT" />
                        <h2 class="subtitle has-text-centered">
                            <div class="content" style="text-align: justify;">
                                <strong>Analysis of gene fusions derived from chimeric read artifacts in
                                    dRNA-seq</strong>
                                <p><strong>(a)</strong> Circos plot depicting chromosomal connections of gene fusions resulting from chimeric read artifacts in VCaP cells. Blue lines represent inter-chromosomal fusion events, while red lines indicate intra-chromosomal
                                    fusions. The outer track displays chromosomal ideograms labeled with respective chromosome numbers.</p>
                                <!-- <p><strong>(b)</strong> GO enrichment analysis of fusion genes derived from chimeric -->
                                <!-- read artifacts identified by DeepChopper in dRNA-seq data from A549, HepG2, and -->
                                <!-- HCT116 cell lines, and VCaP RNA004 dRNA-seq data. The table -->
                                <!-- lists enriched GO terms of biological processes, associated genes, and the -->
                                <!-- statistical significance (p-values) for each enrichment.</p> -->
                            </div>
                        </h2>
                    </div>

                </div>
            </div>
        </div>
    </section>
    <!-- End image carousel -->


    <!-- Video carousel -->

    <!-- <section class="hero is-small">
        <div class="hero-body">
            <div class="container">
                <h2 class="title is-3">Another Carousel</h2>
                <div id="results-carousel" class="carousel results-carousel">
                    <div class="item item-video1">
                        <video poster="" id="video1" autoplay controls muted loop height="100%"> -->

    <!-- Your video file here -->

    <!-- <source src="static/videos/carousel1.mp4"
            type="video/mp4">
          </video>
                    </div>
                    <div class="item item-video2">
                        <video poster="" id="video2" autoplay controls muted loop height="100%"> -->

    <!-- Your video file here -->

    <!-- <source src="static/videos/carousel2.mp4" -->
    <!-- type="video/mp4"> -->
    <!-- </video> -->
    <!-- </div> -->
    <!-- <div class="item item-video3"> -->
    <!-- <video poster="" id="video3" autoplay controls muted loop height="100%">\ -->

    <!-- Your video file here -->

    <!-- <source src="static/videos/carousel3.mp4" -->
    <!-- type="video/mp4"> -->
    <!-- </video> -->
    <!-- </div> -->
    <!-- </div> -->
    <!-- </div> -->
    <!-- </div> -->
    <!-- </section> -->

    <!-- End video carousel -->


    <!-- Paper poster -->
    <!-- <section class="hero is-small is-light">
        <div class="hero-body">
            <div class="container">
                <h2 class="title">Poster</h2>

                <iframe src="static/pdfs/sample.pdf" width="100%" height="550">
          </iframe>

            </div>
        </div>
    </section> -->
    <!--End paper poster -->


    <!--BibTex citation -->
    <section class="section" id="BibTeX">
        <div class="container is-max-desktop content">
            <h2 class="title">BibTeX</h2>
            <pre><code id="bibtex">@article{Li2024.10.23.619929,
    author    = {Li, Yangyang and Wang, Ting-You and Guo, Qingxiang 
                 and Ren, Yanan and Lu, Xiaotong and Cao, Qi 
                 and Yang, Rendong},
    title     = {A Genomic Language Model for Chimera Artifact Detection 
                 in Nanopore Direct RNA Sequencing},
    journal   = {bioRxiv},
    year      = {2024},
    doi       = {10.1101/2024.10.23.619929},
    publisher = {Cold Spring Harbor Laboratory},
    url       = {https://www.biorxiv.org/content/early/2024/10/25/2024.10.23.619929},
    eprint    = {https://www.biorxiv.org/content/early/2024/10/25/2024.10.23.619929.full.pdf}
}</code></pre>
        </div>
    </section>
    <!--End BibTex citation -->


    <footer class="footer">
        <div class="container">
            <div class="columns is-centered">
                <div class="column is-8">
                    <div class="content">

                        <p>
                            This page was built using the <a href="https://github.com/eliahuhorwitz/Academic-project-page-template" target="_blank">Academic Project Page Template</a> which was adopted from the <a href="https://nerfies.github.io" target="_blank">Nerfies</a> project
                            page. <br> This website is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/" target="_blank">Creative
                                Commons Attribution-ShareAlike 4.0 International License</a>.
                        </p>

                    </div>
                </div>
            </div>
        </div>
    </footer>

    <!-- Statcounter tracking code -->

    <!-- You can add a tracker to track page visits by creating an account at statcounter.com -->

    <!-- End of Statcounter Code -->

</body>

</html>
